### ISSUE TITLE
lsp-enable-window causes various problems

### ISSUE THREAD

**Issue by dgmulf** [ul/kak-lsp#220]
Using lsp-enable-window as opposed to lsp-enable causes some syntax highlighting to work incorrectly (e.g., errors are not highlighted selectively), and also prevents lsp-stop from killing the kak-lsp process.

Kakoune v2019.07.01
kak-lsp 6.2.1
python-language-server 0.27.0
Arch Linux

**Comment by ul** [comment]
Would you mind posting a kak-lsp related bit of your kakrc?

**Comment by ul** [comment]
Related: #221

**Comment by ul** [comment]
Please try the latest master, is it any better?

**Comment by dgmulf** [comment]
I just checked out the latest commit (`164abc`), and it does indeed fix the issue of leaving the kak-lsp process hanging around. However, the formatting errors still persist:

![lsp-formatting-bug](https://user-images.githubusercontent.com/5732749/62100868-26366800-b248-11e9-9d05-32d2a11a7b4f.png)

In the above screenshot, the left pane is using
```
eval %sh{kak-lsp --kakoune -s $kak_session}
lsp-enable
```
while the right pane is using
```
eval %sh{kak-lsp --kakoune -s $kak_session}
hook global WinSetOption filetype=(python) %{
    lsp-enable-window
}
```

**Comment by ul** [comment]
This is extremely strange as `lsp-enable-window` uses the same highlighters as `lsp-enable`, just in a different scope... @mawww do you have any ideas about how it's possible for such highlighter breakage/conflict/... to arise?

@dgmulf  I wonder if you get the same diagnostics response in both cases or not? You can enable debug log in kak-lsp as described here https://github.com/ul/kak-lsp/#troubleshooting

**Comment by dgmulf** [comment]
Nope, looks like the debug logs are identical except for timestamps either way.
